### DISCUSSION TITLE
Customize legend title text

### DISCUSSION THREAD

**Discussion by Mengesh** [plantuml-stdlib/C4-PlantUML#363] [Ideas]
Would it make sense to introduce another global variable i.e. `LEGEND_TITLE_TEXT` that would default to `Legend` but also let the user specify it's own title?

**Comment by kirchsth** [comment]
Yes, would be a good idea. It looks like I have overseen it ;-)

I think it would be a small extension.

- line ~57 :add `!$LEGEND_TITLE_TEXT?= "Legend"`
- line ~2131: replace `**Legend**` with `**$LEGEND_TITLE_TEXT **`   (space after $LEGEND_TITLE_TEXT is important that we don't get an error if $LEGEND_TITLE is set with an empty value)

``` 
' Labels
' ##################################

!$LEGEND_TITLE_TEXT?= "Legend"
!$LEGEND_SHADOW_TEXT ?= "shadow"
....
<$colorWithHash(transparent),$colorWithHash(transparent)>|<color:$LEGEND_TITLE_COLOR>**$LEGEND_TITLE_TEXT **</color> |
```

and then it can be used like in (the updated version of) ./percy/TestLegend.puml

```
@startuml

!$LEGEND_TITLE_TEXT = "Updated legend title"
' !$LEGEND_TITLE_TEXT = ""

!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
...
```
![image](https://github.com/user-attachments/assets/077c07ed-d300-42a8-a748-e917e9042f96)

I will try to fix it next weekend.

BR
Helmut

**Comment by kirchsth** [comment]
Hi @Mengesh,

I have a new implementation in MR #366, can you please check it

Thank you and best regards
Helmut

# `UpdateLegendTitle(newTitle)

Based on discussion #363 a new issues was created (#364) and a new  `UpdateLegendTitle(newTitle)` call is added.
This call changes the legen title similar  to the other `Update....(..., ?legendText, ...)` calls which change the legend texts of tags.

Ex. following code:

```plantuml
@startuml
!include https://raw.githubusercontent.com/kirchsth/C4-PlantUML/extended/C4_Container.puml

' NEW CALL UpdateLegendTitle
UpdateLegendTitle("Updated legend title")

Person(admin, "Administrator")
System_Boundary(c1, 'Sample') {
    Container(web_app, "Web Application")
}
System(twitter, "Twitter")

SHOW_LEGEND()
@enduml
``` 

produces following diagram:

[![image](https://github.com/user-attachments/assets/377e9223-0697-489b-9862-a19d302f4abc)](https://www.plantuml.com/plantuml/uml/POz1IyCm5CVl-HHnbsKmMmHFdbPhqKETWsxqMDBaiGRJ92Ilr25-Tr-TU36NF7x_Fp_UsoMK4QVHiXlZb9qqy04nf8SyZtBEpWQ7gPyIHEKTWiDC-J5_Dr4D2OUylByzMEdmz5hdy46v1asmAwaiZOEOXKNDrdnVjRmiwfgVWfO8DPofV3HeWVqZOdL5cjiVo76XgmrZ1uZ9Eo7rQDoMhuhbDmcZH1yfRou9OUmU_UIqZ1UXxhPytSWnM5Xl-2VZzFvM4pFqdGo1F2tql0Z16YNHU4Uchr-NmDaWGgJIyJfHo9gNjxQhg-Tg_oGsR0TEqvNV)

**Reply by kirchsth** [reply]
feature released in [C4-PlantUML v2.11.0](https://github.com/plantuml-stdlib/C4-PlantUML/releases/tag/v2.11.0)
